api/app/controllers/spree/api/v2/storefront/account/addresses_controller.rb

```diff
@@ -26,12 +22,8 @@ def collection
               collection_finder.new(scope: scope, params: params).execute
             end
 
-            def resource
-              @resource ||= scope.find(params[:id])
-            end
-
-            def scope
-              spree_current_user.addresses
+            def model_class
+              Spree::Address
```

Comment:
I would consider using ActiveSupport::Inflections module to avoid hardcoding model class names in each `model_class` method, for example:
`"Spree::#{controller_name.classify}".constantize`

Reply:
@KacperMekarski wanted to avoid that to be more flexible (non-Spree resources controllers) and not to introduce too much "magic"